analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -631,75 +406,129 @@ MET <- fusion_df %>%
+  arrange(desc(Freq))
+```
+
+### H3.3 and IDH wildtype
+
+> * High-grade gliomas absent of H3F3A and IDH mutations
+> * Defining lesions: MYCN, PDGFRA amplification, TP53 and TERT mutations
+
+```{r}
+# which tumors were not yet subtyped?
+not_subtyped <- setdiff(unique(all_data_df$id), molecular_subtype_table$id)
+#"BS_DBH8YWA1"
   
-  select(c(Kids_First_Participant_ID, 
-           sample_id, 
-           starts_with("Kids_First_Biospecimen"), 
-           Kids_First_Biospecimen_ID_RNA, 
-           molecular_subtype, 
-           composition, 
-           HGG_Fusion_evidence, 
-           HGG_Fusion_counts)) %>% 
   
-  filter(molecular_subtype != "delete") %>% 
+wildtype_df <- all_data_df %>%
+  filter(id %in% not_subtyped) %>%
+  left_join(molecular_subtype_table[c("id", "molecular_subtype", "molecular_subtype_methyl")]) %>%
+  filter(`H3F3A_H3-3A.K28M` == "No",
+         HIST1H3B_H3C2.K28M == "No",
+         HIST1H3C_H3C3.K28M == "No",
+         HIST2H3C_H3C14.K28M == "No",
+         `H3F3A_H3-3A.G35R` == "No",
+         `H3F3A_H3-3A.G35V` == "No",
+         IDH1_mutation == "No R132 or R172",
+         # and not re-classified by methyl
+         !molecular_subtype_methyl %in% c("DMG, H3 K28", "DHG, H3 G35", "IHG")) %>%
+  mutate(molecular_subtype = "HGG, H3 wildtype") %>%
+  distinct()
+
+# add wildtype methyl not already there
+wildtype_methyl <- all_data_df %>%
+  filter(!Kids_First_Biospecimen_ID_Methyl %in% wildtype_df$Kids_First_Biospecimen_ID_Methyl,
+         molecular_subtype_methyl == "HGG, H3 wildtype") %>%
+  mutate(molecular_subtype = molecular_subtype_methyl)
+
+wildtype_df <- wildtype_df %>%
+  bind_rows(wildtype_methyl)
+
+wildtype_df %>% arrange(MYCN_focal_status, PDGFRA_focal_status)
+```
+
+### Add samples missing from all_data_df due to not having any mutations/cnv data
+> We joined entire fusion and expression data together earlier, so no RNA specimens should be missing
+> We added non-methyl DNA bs_ids which were not present in CNV or mutation files
+> We added high-confidence methyl bs_ids earlier, so need to add remaining methylation bs_ids
+
+```{r add subtypes if exist for other specimens or the same id and rest as To be classified}
+molecular_subtype_table_wt <- molecular_subtype_table %>%
+  bind_rows(wildtype_df)
+
+# subtype mapping 
+subtype_map <- molecular_subtype_table_wt %>%
+  select(id, molecular_subtype) %>%
   unique()
 
-molecular_subtype_table <- molecular_subtype_table %>% 
-  filter(!sample_id %in% IHG$sample_id) %>%
-  bind_rows(IHG) 
+# which ids have samples which can be assigned subtypes
+all_subtypes <- subtype_map %>%
+  left_join(hgg_meta[,c("Kids_First_Biospecimen_ID", "id")]) %>%
+  select(Kids_First_Biospecimen_ID, id, molecular_subtype)
+
+# add rest as TO BE CLASSIFIED
+to_be_classified <- hgg_meta %>%
+  filter(!id %in% all_subtypes$id) %>%
+  mutate(molecular_subtype = "HGG, To be classified") %>%
+  select(Kids_First_Biospecimen_ID, id, molecular_subtype)
```

Comment:
```
select(Kids_First_Biospecimen_ID, id, molecular_subtype)
```
change to 

```
select(Kids_First_Biospecimen_ID, id, molecular_subtype, Kids_First_Participant_ID, sample_id)
```

Reply:
updated in [87a9ed4](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/331/commits/87a9ed4fc97315ff00995058767888276bdb16db)